yarn-project/p2p/src/services/reqresp/reqresp.test.ts

```diff
@@ -413,6 +426,8 @@ describe('ReqResp', () => {
       const batchSize = 12;
       nodes = await createNodes(peerScoring, 3);
 
+      const requesterLoggerSpy = jest.spyOn((nodes[0].req as any).logger, 'debug');
```

Comment:
I'm not a fan of hooking onto the logger to test for events, but I don't have a better recommendation that doesn't involve extending an `EventEmitter` and firing events that no one except a test cares about, or having an empty virtual `onStuffHappened` method that we extend on a test-only version of the subject.